internal/command/command_service.go

```diff
@@ -184,7 +216,6 @@ func (cs *CommandService) CancelSubscription(ctx context.Context) {
 
 // nolint: revive,gocognit
 func (cs *CommandService) subscribe(ctx context.Context) {
```

Comment:
Looks like the adjustments I made to Donal's PR didn't make it in here?
https://github.com/nginx/agent/blob/5f289815d11e5e6ef9852e25982db13abc202763/internal/command/command_service.go#L181-L207

The issue was the repeated locking in the for loop.